xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/ListenerProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.listener;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.listener.chaining.ChainingListener;
+import org.xwiki.rendering.listener.chaining.ListenerChain;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Allows to access a listener instance.
+ *
+ * @version $Id$
+ * @since 15.3RC1
+ * @since 14.10.8
+ */
+@Role
+@Unstable
+public interface ListenerProvider
+{
+    /**
+     * @param action the action performed by the listener ("render" or "parse")
+     * @param syntaxHint the hint of the syntax using for the action
+     * @return {@code true} when the listener provider can return a listener for the given action and syntaxHint
+     */
+    boolean accept(String action, String syntaxHint);
```

Comment:
Why not pass a Syntax ?

Reply:
Indeed, might be cleaner. The only cons is that we now need to parse the hints.

Reply:
Done